### PR TITLE
Remove RichColumnDescriptor from parquet module

### PR DESCRIPTION
**Pull request by charlesjmorgan**
## Description

The `RichColumnDescriptor` class in the parquet module seems a bit redundant. This PR removes that class from the `PrimitiveField` and replaces the columns field in the `ParquetReader` with a list of `PrimitiveField`'s

> Is this change a fix, improvement, new feature, refactoring, or other?

refactor

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Changes in parquet library and the update usages in hive, and iceberg

> How would you describe this change to a non-technical end user or system administrator?

Remove a redundant class

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
```

### CODE REVIEW

**Review by skrzypo987 — APPROVED** [review]

File: lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 package io.trino.parquet;
```
Comment (sopel39) [inline lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java:14]: You didn't remove `io.trino.parquet.RichColumnDescriptor` class. Is it still used?

Could you improve commit description? Why we don't need `io.trino.parquet.RichColumnDescriptor` anymore

**Review by sopel39 — COMMENTED** [review]
lgtm % comment

File: lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java (hunk 1 above)
Comment (charlesjmorgan) [reply lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java:14]: Huh I thought I had removed it, it has been removed now. Commit message has been updated as well.

**Review by sopel39 — APPROVED** [review]

File: lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java [hunk 2]
```diff
@@ -112,8 +109,8 @@
     private long nextRowInGroup;
     private int batchSize;
     private int nextBatchSize = INITIAL_BATCH_SIZE;
-    private final PrimitiveColumnReader[] columnReaders;
-    private final long[] maxBytesPerCell;
+    private final Map<Integer, PrimitiveColumnReader> columnReaders;
```
Comment (sopel39) [inline lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java:112]: Can this cause performance regression? Can it be primitive list?

**Pull request merged**